src/Chess/ChessBoard.hs

```diff
@@ -15,11 +16,114 @@ freshBoard = fenToChessBoard "rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR"
+                        ++ "\n"
+                        ++ "(2)  "  ++ retrievePieceAt ('A',2) chessBoard
+                                    ++ retrievePieceAt ('B',2) chessBoard 
+                                    ++ retrievePieceAt ('C',2) chessBoard 
+                                    ++ retrievePieceAt ('D',2) chessBoard 
+                                    ++ retrievePieceAt ('E',2) chessBoard 
+                                    ++ retrievePieceAt ('F',2) chessBoard 
+                                    ++ retrievePieceAt ('G',2) chessBoard 
+                                    ++ retrievePieceAt ('H',2) chessBoard 
+                        
+                        ++ "\n"
+                        ++ "(1)  "  ++ retrievePieceAt ('A',1) chessBoard
+                                    ++ retrievePieceAt ('B',1) chessBoard 
+                                    ++ retrievePieceAt ('C',1) chessBoard 
+                                    ++ retrievePieceAt ('D',1) chessBoard 
+                                    ++ retrievePieceAt ('E',1) chessBoard 
+                                    ++ retrievePieceAt ('F',1) chessBoard 
+                                    ++ retrievePieceAt ('G',1) chessBoard 
+                                    ++ retrievePieceAt ('H',1) chessBoard 
+                        
+                        ++ "\n\n" 
+
+retrievePieceAt:: ChessPosition -> ChessBoard -> [Char]
+retrievePieceAt (char,num) board
+    | getPieceAt (char,num) board == Nothing                    = empty
+    | getPieceAt (char,num) board == Just (King White)          = whiteKing
+    | getPieceAt (char,num) board == Just (Queen White)         = whiteQueen
+    | getPieceAt (char,num) board == Just (Rook White)          = whiteRook
+    | getPieceAt (char,num) board == Just (Bishop White)        = whiteBishop
+    | getPieceAt (char,num) board == Just (Knight White)        = whiteKnight
+    | getPieceAt (char,num) board == Just (Pawn White)          = whitePawn
+    | getPieceAt (char,num) board == Just (King Black)          = blackKing
+    | getPieceAt (char,num) board == Just (Queen Black)         = blackQueen
+    | getPieceAt (char,num) board == Just (Rook Black)          = blackRook
+    | getPieceAt (char,num) board == Just (Bishop Black)        = blackBishop
+    | getPieceAt (char,num) board == Just (Knight Black)        = blackKnight
+    | getPieceAt (char,num) board == Just (Pawn Black)          = blackPawn
+    where
+            empty = "[ ]"
+            whiteKing = "[\x2654]"
+            whiteQueen = "[\x2655]"
+            whiteRook = "[\x2656]"
+            whiteBishop = "[\x2657]"
+            whiteKnight = "[\x2658]"
+            whitePawn = "[\x2659]"
+
+            blackKing = "[\x265A]"
+            blackQueen = "[\x265B]"
+            blackRook = "[\x265C]"
+            blackBishop = "[\x265D]"
+            blackKnight = "[\x265E]"
+            blackPawn = "[\x265F]"
```

Comment:
Here's a shorter way to write this. I would also change the name to `getPieceAsString` and probably move it to `ChessPieces.hs`. I wasn't able to test this though because I couldn't get your tests to run, so you'll have to double check that.
```suggestion
chessPieceStrings = [((King   White), "[\x2654]")
                    ,((Queen  White), "[\x2655]")
                    ,((Rook   White), "[\x2656]")
                    ,((Bishop White), "[\x2657]")
                    ,((Knight White), "[\x2658]")
                    ,((Pawn   White), "[\x2659]")
                    ,((King   Black), "[\x265A]")
                    ,((Queen  Black), "[\x265B]")
                    ,((Rook   Black), "[\x265C]")
                    ,((Bishop Black), "[\x265D]")
                    ,((Knight Black), "[\x265E]")
                    ,((Pawn   Black), "[\x265F]")]
getPieceAsString:: ChessPosition -> ChessBoard -> [Char]
getPieceAsStringposition board = if piece /= Nothing then fromJust (lookup (fromJust piece) chessPieceStrings) else "[ ]"  where piece = getPieceAt position board
```